store/key_value_store_server.cc

```diff
@@ -35,12 +34,57 @@ bool KvstoreServiceImpl::fileExists(const char *fileName)
 }
 
 void KvstoreServiceImpl::CreateKeyFile(const std::string &key, const std::string &value){
+  if (!fileExists("key-pair.txt")){
+    std::ofstream myfile;
+    myfile.open("key-pair.txt", std::ios_base::app);
+    myfile << key << ":" << value << "\n";
+    myfile.close();
+  } else {
+    std::fstream infile("key-pair.txt");
+    std::string line;
+    std::string delimiter = ":";
+    int i = 0;
+    const char * v = value.c_str();
+    const char * k = key.c_str();
+    while(std::getline(infile, line)){
+      i++;
+      std::string first = line.substr(0,line.find(delimiter));
+      std::string second = line.substr(line.find(delimiter)+1, line.length());
+      const char * f = first.c_str();
+      if (strcmp(k,f)==0){
+        LOG(INFO) << "FOUND SAME USER";
+        Deleteline("key-pair.txt", i);
+        Writeline("key-pair.txt",k, v);
+        return;
+      }
+    }
+    Writeline("key-pair.txt",k, v); 
+  }
 }
 
 void KvstoreServiceImpl::Writeline(const char *file_name, const char *key, const char *val){
+  std::ofstream ofs;
+  ofs.open(file_name, std::ios_base::app);
+  ofs << key << ":" << val << "\n";
+  ofs.close();
 }
 
 void KvstoreServiceImpl::Deleteline(const char *file_name, int n) { 
+    std::ifstream is(file_name); 
+    std::ofstream ofs; 
+    ofs.open("temp.txt", std::ofstream::out); 
+    char c; 
+    int line_no = 1; 
+    while (is.get(c)) { 
+        if (c == '\n') 
+        line_no++; 
+        if (line_no != n) 
+            ofs << c; 
+    } 
+    ofs.close(); 
+    is.close(); 
+    std::remove(file_name); 
+    std::rename("temp.txt", file_name); 
```

Comment:
I am unclear on the remove source code, though I could see this operation getting costly with larger files.